src/builtins/core/date.rs

```diff
@@ -490,6 +490,25 @@ impl PlainDate {
         Self::try_new(date.year, date.month, date.day, calendar)
     }
 
+    /// Converts a UTF-16 encoded string into a `PlainDate`.
+    pub fn from_utf16(s: &[u16]) -> TemporalResult<Self> {
```

Comment:
n.b. temporal_capi has a bunch of internally-converting `from_utf16`s that probably can be swapped over to these